From 9b5567d849e49edc3e008522c3598f5138a8f753 Mon Sep 17 00:00:00 2001
From: Hao Bui <hao.bui.yg@renesas.com>
Date: Tue, 13 Apr 2021 14:57:03 +0700
Subject: [PATCH 428/433] ARM: dts: rz/g1: Add support for OV7725 camera

Add support for OV7725 CMOS camera and mark it as disabled.
Note that OV5640 is enabled by default in RZ/G1[H|M|N|E] platforms and
OV7725 can only be enabled via U-boot command line as below:

	iWave-G2x> setenv vin2_camera 'ov7725'

This command will disable OV5640 and replace it by OV7725.

Signed-off-by: Hao Bui <hao.bui.yg@renesas.com>
---
 arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi        | 20 +++++++
 arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts | 80 +++++++++++++++++++++++++
 arch/arm/boot/dts/r8a7745-iwg22d-sodimm.dts     | 20 +++++++
 3 files changed, 120 insertions(+)

diff --git a/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi b/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi
index 1629737..84f6fdb 100644
--- a/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi
+++ b/arch/arm/boot/dts/iwg20d-q7-dbcm-ca.dtsi
@@ -144,6 +144,26 @@
 			};
 		};
 	};
+
+	ov7725@21 {
+		compatible = "ov772x";
+		reg = <0x21>;
+		remote = <&vin2>;
+
+		/* mark OV7725 as disabled by default
+		 * to enable, run command below under U-boot command line
+		 * 	setenv vin2_camera 'ov7725'
+		 * Note: no need to update status here
+		 */
+		status = "disabled";
+
+		port {
+			ov7725: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin2ep>;
+			};
+		};
+	};
 };
 
 &pfc {
diff --git a/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts b/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts
index a3ac85e..0370346 100644
--- a/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts
+++ b/arch/arm/boot/dts/r8a7742-iwg21d-q7-dbcm-ca.dts
@@ -79,6 +79,26 @@
 			};
 		};
 	};
+
+	ov7725@0 {
+		compatible = "ov772x";
+		reg = <0x21>;
+		remote = <&vin0>;
+
+		/* mark OV7725 as disabled by default
+		 * to enable, run command below under U-boot command line
+		 * 	setenv vin0_camera 'ov7725'
+		 * Note: no need to update status here
+		 */
+		status = "disabled";
+
+		port {
+			ov7725_0: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin0ep>;
+			};
+		};
+	};
 };
 
 &i2c1 {
@@ -100,6 +120,26 @@
 			};
 		};
 	};
+
+	ov7725@1 {
+		compatible = "ov772x";
+		reg = <0x21>;
+		remote = <&vin1>;
+
+		/* mark OV7725 as disabled by default
+		 * to enable, run command below under U-boot command line
+		 * 	setenv vin1_camera 'ov7725'
+		 * Note: no need to update status here
+		 */
+		status = "disabled";
+
+		port {
+			ov7725_1: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin1ep>;
+			};
+		};
+	};
 };
 
 &i2c2 {
@@ -115,6 +155,26 @@
 			};
 		};
 	};
+
+	ov7725@2 {
+		compatible = "ov772x";
+		reg = <0x21>;
+		remote = <&vin2>;
+
+		/* mark OV7725 as disabled by default
+		 * to enable, run command below under U-boot command line
+		 * 	setenv vin2_camera 'ov7725'
+		 * Note: no need to update status here
+		 */
+		status = "disabled";
+
+		port {
+			ov7725_2: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin2ep>;
+			};
+		};
+	};
 };
 
 &i2c3 {
@@ -136,6 +196,26 @@
 			};
 		};
 	};
+
+	ov7725@3 {
+		compatible = "ov772x";
+		reg = <0x21>;
+		remote = <&vin3>;
+
+		/* mark OV7725 as disabled by default
+		 * to enable, run command below under U-boot command line
+		 * 	setenv vin3_camera 'ov7725'
+		 * Note: no need to update status here
+		 */
+		status = "disabled";
+
+		port {
+			ov7725_3: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin3ep>;
+			};
+		};
+	};
 };
 
 
diff --git a/arch/arm/boot/dts/r8a7745-iwg22d-sodimm.dts b/arch/arm/boot/dts/r8a7745-iwg22d-sodimm.dts
index 82f04aa..e6eb7a1 100644
--- a/arch/arm/boot/dts/r8a7745-iwg22d-sodimm.dts
+++ b/arch/arm/boot/dts/r8a7745-iwg22d-sodimm.dts
@@ -221,6 +221,26 @@
 			};
 		};
 	};
+
+	ov7725@21 {
+		compatible = "ov772x";
+		reg = <0x21>;
+		remote = <&vin1>;
+
+		/* mark OV7725 as disabled by default
+		 * to enable, run command below under U-boot command line
+		 * 	setenv vin1_camera 'ov7725'
+		 * Note: no need to update status here
+		 */
+		status = "disabled";
+
+		port {
+			ov7725: endpoint {
+				bus-width = <8>;
+				remote-endpoint = <&vin1ep>;
+			};
+		};
+	};
 };
 
 &pci0 {
-- 
2.7.4

